src/travertino/node.py

```diff
@@ -14,6 +18,71 @@ def __init__(self, style, applicator=None, children=None):
             for child in children:
                 self.add(child)
 
+    @property
+    def style(self):
+        """The node's style.
+
+        Assigning a style triggers an application of that style if an applicator has
+        already been assigned.
+        """
+
+        return getattr(self, "_style", None)
+
+    @style.setter
+    def style(self, style):
+        if style is None:
+            self._style = None
+            return
+
+        self._style = style.copy()
+        self.intrinsic = self.style.IntrinsicSize()
+        self.layout = self.style.Box(self)
+
+        if self.applicator:
+            self.style._applicator = self.applicator
+
+            try:
+                self.style.reapply()
+            # Backwards compatibility for Toga
+            except AttributeError:
+                warn(
+                    "Failed to apply style, when new style assigned and applicator "
+                    "already present. Node should be sufficiently initialized to "
+                    "apply style before it has both a style and an applicator.",
+                    DeprecationWarning,
+                    stacklevel=2,
+                )
+
+    @property
+    def applicator(self):
+        """This node's applicator, which handles applying the style.
+
+        Assigning an applicator triggers an application of that style if a style has
+        already been assigned.
+        """
+        return getattr(self, "_applicator", None)
+
+    @applicator.setter
+    def applicator(self, applicator):
+        self._applicator = applicator
+
+        if self.style is not None:
+            self.style._applicator = applicator
+
+            if applicator:
+
+                try:
+                    self.style.reapply()
+                # Backwards compatibility for Toga
+                except AttributeError:
+                    warn(
+                        "Failed to apply style, when new applicator assigned and style "
+                        "already present. Node should be sufficiently initialized to "
+                        "apply style before it has both a style and an applicator.",
+                        DeprecationWarning,
+                        stacklevel=2,
+                    )
```

Comment:
I share your ick-factor with this. 

My only thought on de-icking it would actually be to make the handling *more* generic - catch *any* `Exception` when the style is re-applied, and generate a general API warning, rather than a Deprecation warning. After all, the issue being flagged is that it wasn't possible to apply the style to the node, which is a general requirement of the API now. That isn't really a deprecation issue - it's a hardening of API requirements.

Reply:
That makes sense. Do you think the message should perhaps include a note that this may be an exception in future?

Reply:
Warning about a future exception would make sense to me.